hs-bindgen-runtime/src/HsBindgen/Runtime/CEnum.hs

```diff
@@ -151,44 +164,51 @@ class CEnum a => SequentialCEnum a where
 -- An empty list is returned when the specified value is not declared.
 getNames :: forall a. CEnum a => a -> [String]
 getNames x = maybe [] NonEmpty.toList $
-    Map.lookup (fromCEnum x) (getDeclaredValues (Proxy :: Proxy a))
+    Map.lookup (fromCEnum x) (getIntegralToDeclaredValues (Proxy :: Proxy a))
 
 {-------------------------------------------------------------------------------
   Instance support
 -------------------------------------------------------------------------------}
 
--- | Declared values and associated names (opaque)
-newtype DeclaredValues a = DeclaredValues
-    { unDeclaredValues :: Map (CEnumZ a) (NonEmpty String)
+-- | Declared values (opaque)
+data DeclaredValues a = DeclaredValues
+    { declaredValuesList :: ![String]
```

Comment:
A bang on a list is a red flag to me (it forces only the head of the list, which is almost certainly not what we want). To avoid delaying this PR further I'll just fix this myself, by removing this field altogether and adding

```haskell
declaredValuesList :: DeclaredValues a -> [String]
declaredValuesList = Map.keys . declaredValueToIntegral
```